daemon/top_unix.go

```diff
@@ -79,6 +79,10 @@ func parsePSOutput(output []byte, procs []uint32) (*container.ContainerTopOKBody
 	if pidIndex == -1 {
 		return nil, fmt.Errorf("Couldn't find PID field in ps output")
 	}
+	//remove appended pid in first cloumn
+	if blHaso {
+		procList.Titles = procList.Titles[1:]
```

Comment:
You can do something like

At the beginning of the function:

```go
skipCol := 0
if pidAdded {
         skipCol = 1
}
```

and everywhere later in in the code, for example
```go
 	procList.Titles = fieldsASCII(lines[0])[skipCol:]
```
and
```go
	appendProcess2ProcList(procList, fields[skipCol:])
￼```